src/main/java/com/minecolonies/coremod/client/render/worldevent/ColonyBlueprintRenderer.java

```diff
@@ -395,8 +396,11 @@ public Map<BlockPos, PendingRenderData> getDesiredBlueprints(final WorldEventCon
                     {
                         String schemPath = buildingView.getStructurePath();
                         schemPath = schemPath.replace(".blueprint", "");
-                        schemPath = schemPath.substring(0, schemPath.length() - 1) + buildingView.getBuildingMaxLevel() + ".blueprint";
-
+                        try{
+                            schemPath = schemPath.substring(0, schemPath.length() - 1) + buildingView.getBuildingMaxLevel() + ".blueprint";

```

Comment:
If it crashes here its probably because schemPath is empty. So we can do an isEmpty check and add some useful logging.

Reply:
Also, just noticed that this entire section has not had the formatter run over it.